contrib/coreos/user-data.example

```diff
@@ -190,58 +209,67 @@ write_files:
     permissions: '0755'
     content: |
       #!/usr/bin/bash
-      if [ "$1" != '--really' ]; then
-        echo "command must be run as: $0 --really"
-        exit 1
-      fi
-      # procedure requires the store-admin
-      ADMIN_RUNNING=$(docker inspect --format="{{ .State.Running }}" deis-store-admin)
-      if [ $? -eq 1 ] || [ "$ADMIN_RUNNING" == "false" ]; then
-        echo "deis-store-admin container is required for graceful shutdown"
-        exit 2
-      fi
-      set -e -x -o pipefail
-      # determine osd id
-      CURRENT_STATUS=$(docker exec deis-store-admin ceph health | awk '{print $1}')
-      OSD_HOSTS=($(etcdctl ls /deis/store/hosts/| awk -F'/' '{print $5}'))
-      for HOST in "${OSD_HOSTS[@]}"
-      do
-        PUBLIC_IP=$(fleetctl list-machines -fields="machine,ip" -full -no-legend| grep `cat /etc/machine-id` | awk '{print $2}')
-        if [ "$HOST" = "$PUBLIC_IP" ] ; then
-          OSD_ID=$(etcdctl get /deis/store/osds/$PUBLIC_IP)
-          break
-        fi
-      done
-      # if we own an osd and its healthy, try to gracefully remove it
-      if [ ! -z "$OSD_ID" ] && [[ "$CURRENT_STATUS" == *"HEALTH_OK"* ]] && [ ${#OSD_HOSTS[@]} -gt "3" ]; then
-        docker exec deis-store-admin ceph osd out $OSD_ID
-        sleep 30
-        TIMEWAITED=0
-        until [[ $(docker exec deis-store-admin ceph health) == *"HEALTH_OK"* ]]
+      ceph_shutdown () {
+        # determine osd id
+        OSD_HOSTS=($(etcdctl ls /deis/store/hosts/| awk -F'/' '{print $5}'))
+        for HOST in "${OSD_HOSTS[@]}"
         do
-          if [ $TIMEWAITED -gt "1200" ]
-          then
-            echo "ceph graceful removal timeout exceeded"
+          PUBLIC_IP=$(etcdctl member list|grep `cat /etc/machine-id`| awk '{print $3}'| grep -Eo '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}')
```

Comment:
You could do `etcdctl get /deis/store/osds/$(hostname -i)` (or `--ip-address` for clarity) instead of looping through `etcdctl member list`. Then you are just checking for `$?` success to get the `OSD_ID`

Not entirely sure it would work the way we expect if we throw in some isolation.